cpp/src/arrow/filesystem/s3fs_test.cc

```diff
@@ -1579,5 +1612,26 @@ TEST(S3GlobalOptions, DefaultsLogLevel) {
   }
 }
 
+TEST(CalculateSSECKeyMD5, Sanity) {
+  ASSERT_FALSE(CalculateSSECKeyMD5("").ok());  // invalid length
+  ASSERT_FALSE(CalculateSSECKeyMD5("1234567890123456789012345678901234567890")
+                   .ok());  // invalid length
+  // valid case, with ascii input key
+  ASSERT_OK_AND_ASSIGN(auto result,
+                       CalculateSSECKeyMD5("NLbTMHZn9aCW3Li3ViAdBsoIldPCREw1"))
+  ASSERT_STREQ(result.c_str(),
+               "ZWMxZGUyOTZhMDQwZWJmM2Q4N2E4ZTczZjdlNmI3Mzk=");  // valid case
+  // valid case, with some non-ASCII character and a null byte in the sse_customer_key
```

Comment:
Do we need ASCII case?
It seems that non-ASCII case is only enough.

Reply:
Could you check this?

Reply:
is there any side-effect to include both the ascii and non-ascii case? as my understanding, non-ascii case would like a case to test whether the CalculateSSECustomerKeyMD5 function handle the binary well( in case there are some inproper handling depends on the input string is null terminated) while the ascii case would like the happy path cover.

Reply:
Key is binary data, right?
I think that ASCII only case is very rare case.
And non-ASCII case covers ASCII only case.

Reply:
fine, i could remove the ascii only case.